onboarding-site/entrypoint.sh

```diff
@@ -24,4 +24,11 @@ cat >/var/www/localhost/htdocs/device-info.js <<EOF
   }
 EOF
 cd /var/www/localhost/htdocs
-../busybox httpd -f -p 85
+../busybox httpd -f -p 85 &
+BUSYBOX_PID=$!
+# Trick to catch failures: Wait a few seconds for busybox to exit. If it does,
+# there's likely an error, and the kill will fail, which `set -e` will handle.
+sleep 3
+kill -0 $BUSYBOX_PID
```

Comment:
-_- Is this really the best way to catch failures on startup?

Reply:
It is the only way I've found. It's tricky to both start it in the background, and catch errors from it. If you have other suggestions I'm all ears.

Reply:
Can we not create a service file for the server, which relies on the entrypoint service?

Reply:
I agree that's a better solution in general. But I think I will save myself the work here, since this is not a production service, and `entrypoint.sh` is also used for demoing the page locally. This would break if I added two separate services. Not unsolvable, but too much effort IMHO.

Reply:
Ur the boss, boss

Reply:
It bugged me that it feels like you're right, so I investigated some more. :-) However, I realized that it won't work with your suggestion, because busybox itself has no way of letting systemd know that it is successfully running. Systemd would just exec the process successfully and conclude it's working, even if it would fail milliseconds later because of the port being busy. So eventually you would then end up with the same hack as I already did.

The two ways to do it properly would either be:

* Busybox has to report that it is running (infeasible, it has no support for that)
* We can try to connect to the port to check that busybox is running there (possible, but I would rather not add a dependency on `nc`, which can be missing from Yocto images)

So I think that's the final nail in the coffin.

Reply:
After reading a bit about this, I think this is just fine. But why not curl?

Reply:
Yocto minimal RPi3 image:

```
# curl
-sh: curl: command not found
```

Reply:
:cry:  You win

Reply:
However, busybox has `netcat` bundled.

Reply:
Another counter: I actually disagree with my own suggestion of using `nc`: If busybox fails because the port is busy, then using `nc` will succeed, but busybox will still fail, and whatever is on the port won't be the correct content. Hence using process death as an indicator is still the better approach. :-)

Reply:
haha, I give up!